src/Rule/ForbidAssignmentNotMatchingVarDocRule.php

```diff
@@ -95,6 +100,16 @@ public function processNode(Node $node, Scope $scope): array
         $valueTypeString = $valueType->describe(VerbosityLevel::precise());
         $varPhpDocTypeString = $variableType->describe(VerbosityLevel::precise());
 
+        if ($valueType->accepts($variableType, $scope->isDeclareStrictTypes())->yes() && $variableType->isArray()->no()) {
```

Comment:
I'm thinking about the array exclusion, technically PHPStan is correct even with arrays.

```
Cannot narrow array{id: int, value: string} to array{id: int, value: string, notPresent: bool}
Cannot narrow array{id: mixed, value: mixed} to array{id: int, value: string, notPresent: bool}
```

Generally I try to avoid those hidden conditions in code as it need to be documented and there need to be very good reason for it.

Let me think about that a for a moment..

Reply:
Well in case of array it is not exactly narrowing.

Reply:
Yeah, I thought about using diffferent word for it, but I did not think up any better name. I was also thinking about whitelisting "narrowable" types instead of blacklisting "non-narrowable", but that seems hard too. 

So lets finalize this as it is. Thanks